stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -647,11 +657,12 @@
 ;;   * amount-ustx: the total amount of ustx the delegate may be allowed to lock
 ;;   * until-burn-ht: an optional burn height at which this delegation expires
 ;;   * pox-addr: an optional address to which any rewards *must* be sent
+;;   * signer-key: an optional signer key, that when set the delegate must use.
 (define-public (delegate-stx (amount-ustx uint)
                              (delegate-to principal)
                              (until-burn-ht (optional uint))
-                             (pox-addr (optional { version: (buff 1),
-                                                   hashbytes: (buff 32) })))
+                             (pox-addr (optional { version: (buff 1), hashbytes: (buff 32) }))
+                             (signer-key (optional (buff 33))))
```

Comment:
Can you link the exact discussion? I see https://github.com/stacks-network/stacks-core/pull/4092#discussion_r1409826088

Jude's comment makes it sound like we'd want the Stacker delegating to be able signal a signing key it wants to see for its lockup just like how it can do it with a pox address. If the user signals a different key then the delegate would not call `delegate-stack-stx`.

Reply:
https://github.com/stacks-network/stacks-core/discussions/4082#discussioncomment-7693307

Reply:
There should not be an argument for a signing key in `delegate-stx`

Reply:
I still don't see why not. Let's finish on #4082

Reply:
Settled with https://github.com/stacks-network/stacks-core/discussions/4082#discussioncomment-7931963